test/system/src/main/java/io/pravega/test/system/framework/services/docker/PravegaSegmentStoreDockerService.java

```diff
@@ -0,0 +1,143 @@
+        //set env
+        String zk = "zookeeper:" + ZKSERVICE_ZKPORT;
+        String con = "controller:" + CONTROLLER_PORT;
+        //System properties to configure SS service.
+        String hostSystemProperties =
+                setSystemProperty("autoScale.muteInSeconds", "120") +
+                setSystemProperty("autoScale.cooldownInSeconds", "120") +
+                setSystemProperty("autoScale.cacheExpiryInSeconds", "120") +
+                setSystemProperty("autoScale.cacheCleanUpInSeconds", "120") +
+                setSystemProperty("log.level", "DEBUG") +
+                setSystemProperty("curator-default-session-timeout", String.valueOf(30 * 1000))+
+                        setSystemProperty("hdfs.replaceDataNodesOnFailure", "false");
+
+        String env1 = "PRAVEGA_SEGMENTSTORE_OPTS=" + hostSystemProperties;
+        String env2 = "JAVA_OPTS=-Xmx900m";
+        String env3 = "ZK_URL=" + zk;
+        String env4 = "BK_ZK_URL=" + zk;
+        String env5 = "CONTROLLER_URL=" + con;
+        List<String> stringList = new ArrayList<>();
+        stringList.add(env1);
+        stringList.add(env2);
+        getCustomEnvVars(stringList, SEGMENTSTORE_EXTRA_ENV);
+        stringList.add(env3);
+        stringList.add(env4);
+        stringList.add(env5);
+
+        final TaskSpec taskSpec = TaskSpec
+                .builder()
+                .containerSpec(ContainerSpec.builder().image(IMAGE_PATH + "/nautilus/pravega:" + PRAVEGA_VERSION)
+                        .healthcheck(ContainerConfig.Healthcheck.create(null, 1000000000L, 1000000000L, 3))
+                        .mounts(Arrays.asList(mount))
+                        .env(stringList).args("segmentstore").build())
+                .resources(ResourceRequirements.builder()
+                        .reservations(Resources.builder()
+                                .memoryBytes(setMemInBytes(mem)).nanoCpus(setNanoCpus(cpu)).build())
+                        .build())
+                .build();
+        ServiceSpec spec = ServiceSpec.builder().name(serviceName).taskTemplate(taskSpec).mode(ServiceMode.withReplicas(instances))
+                .networks(NetworkAttachmentConfig.builder().target("docker-network").build())
+                .endpointSpec(EndpointSpec.builder().addPort(PortConfig.builder().
+                        targetPort(SEGMENTSTORE_PORT).protocol("TCP").build())
+                        .build()).build();
+        return spec;
+    }
+
+    private void getCustomEnvVars(List<String> stringList, String segmentstoreExtraEnv) {
+        log.info("Extra segment store env variables are {}", segmentstoreExtraEnv);
+        if (!Strings.isNullOrEmpty(segmentstoreExtraEnv)) {
+            Arrays.stream(segmentstoreExtraEnv.split(ENV_SEPARATOR)).forEach(str -> {
+                String[] pair = str.split(KEY_VALUE_SEPARATOR);
+                if (pair.length != 2) {
+                    log.warn("Key Value not present {}", str);
```

Comment:
throw?

Reply:
throwing `RunTimeException`, done.

Reply:
Why is throwing a generic `RuntimeException` right here? Should it be more specific? Note that I'm not following the upstream flow, so I don't know how this is being handled.

Reply:
OK, I followed that paths through which the exception will bubble up, and it seems fine.